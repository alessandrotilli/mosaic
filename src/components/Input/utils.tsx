import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
