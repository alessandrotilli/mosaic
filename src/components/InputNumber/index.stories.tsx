import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, select, number, text } from "@storybook/addon-knobs";
import { InputSize, InputVariant } from "../../types/Input";
import FormMock from "../../utils/mocks/FormMock";
import IntlProviderMock, { LocaleMock, MessageMock } from "../../utils/mocks/IntlProviderMock";
import { getDocumentationPage, StoriesWrapper } from "../../utils/stories";
import InputNumber, { InputNumberWithProps, DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS } from ".";

export default {
  title: "InputNumber",
  component: InputNumberWithProps,
  parameters: {
    ...getDocumentationPage({
      basedOn: "@material-ui/core/TextField",
      component: "InputNumber",
      e2eTestInfo: {
        dataCyDefault: DATA_CY_DEFAULT,
        dataCyShortcut: DATA_CY_SHORTCUT,
      },
      localizableProps: LOCALIZABLE_PROPS,
    }),
  },
};

export const Canvas = () => (
  // FormMock simulates external form component handling state
  // In a real case scenario "onChange" and "value" props must be passed to InputNumber
  <FormMock inputValue={number("value", 5)} onInputChange={action("onChange callback")}>
    <InputNumber
      dataCy={text("dataCy", DATA_CY_DEFAULT)}
      disabled={boolean("disabled", false)}
      integer={boolean("integer", true)}
      label={text("label", "Label")}
      localized={boolean("localized", false)}
      minValue={number("minValue", 0)}
      maxValue={number("maxValue", 9)}
      onChange={action("onChange callback")}
      placeholder={text("placeholder", "Placeholder Text")}
      required={boolean("required", false)}
      shrink={boolean("shrink", true)}
      size={select("size", InputSize, InputSize.default)}
      variant={select("variant", InputVariant, InputVariant.default)}
      value={number("value", 5)}
    />
  </FormMock>
);

export const Bounded = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Bounded 0-9" minValue={0} maxValue={9} required value={5} />
  </StoriesWrapper>
);

export const DefaultNull = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Default null" />
  </StoriesWrapper>
);

export const IntegerAndFloat = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Integer" value={5} />
    <InputNumber dataCy="input-number" integer={false} label="Float" value={5.5} />
  </StoriesWrapper>
);

export const Localized = () => (
  // IntlProviderMock simulates external IntlProvider context
  <StoriesWrapper>
    <IntlProviderMock locale={select("locale", LocaleMock, LocaleMock.en)}>
      <InputNumber
        label={MessageMock.inputNumber}
        localized
        onChange={action("Click on Button")}
        placeholder={MessageMock.inputNumber}
      />
    </IntlProviderMock>
  </StoriesWrapper>
);

export const Required = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Required" required value={10} />
  </StoriesWrapper>
);

export const Size = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Default" value={10} />
    <InputNumber dataCy="input-number" label="Small" size={InputSize.small} value={10} />
  </StoriesWrapper>
);

export const Variant = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Default" value={10} />
    <InputNumber dataCy="input-number" label="Filled" variant={InputVariant.filled} value={10} />
  </StoriesWrapper>
);
