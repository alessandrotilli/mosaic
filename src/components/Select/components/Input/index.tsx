import React, { CSSProperties, FC, useMemo } from "react";
import {
  AutocompleteRenderInputParams as MUIAutocompleteRenderInputParams,
  Skeleton as MUISkeleton,
  TextField as MUITextField,
} from "@mui/material";

import { ILoadable } from "../../../../types/Base";
import { IInputField } from "../../../../types/Input";
import { getComposedDataCy, ISubpart } from "../../../../utils";

type ISelectInputField = Pick<IInputField, "dataCy" | "label" | "placeholder"> &
  Required<Pick<IInputField, "required" | "size" | "type" | "variant">>;

interface ISelectInput extends ISelectInputField, ILoadable {
  forwarded: MUIAutocompleteRenderInputParams;
}

export const SELECT_LOADING_SUBPART: ISubpart = {
  label: "Loading",
};

const SelectInput: FC<ISelectInput> = ({
  dataCy = "select-input",
  forwarded,
  label,
  loading,
  placeholder,
  required,
  size,
  style: externalStyle,
  type,
  variant,
}) => {
  const { inputProps: forwardedInputProps, ...forwardedRefs } = forwarded;

  const style = useMemo((): CSSProperties => ({ ...externalStyle, width: "100%" }), [externalStyle]);

  const inputDataCy = useMemo(
    () => (!loading ? dataCy : getComposedDataCy(dataCy, SELECT_LOADING_SUBPART)),
    [dataCy, loading]
  );

  const inputProps = useMemo(
    () => ({
      ...forwardedInputProps,
      "data-cy": inputDataCy,
      style,
    }),
    [forwardedInputProps, inputDataCy, style]
  );

  if (loading) {
    return (
      <MUISkeleton width="100%">
        <MUITextField
          {...forwardedRefs}
          inputProps={inputProps}
          margin="normal"
          size={size}
          variant={variant}
          style={style}
        />
      </MUISkeleton>
    );
  }

  return (
    <MUITextField
      {...forwardedRefs}
      inputProps={inputProps}
      label={label}
      margin="normal"
      placeholder={placeholder}
      required={required}
      size={size}
      style={style}
      type={type}
      variant={variant}
    />
  );
};

export default SelectInput;
