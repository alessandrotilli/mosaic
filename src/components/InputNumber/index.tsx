import React, { FC } from "react";
import { InputDataType, InputSize, InputVariant } from "../../types/Input";
import { IInputNumber } from "../../types/InputNumber";
import { StyledMUITextField } from "./styled";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";

const getControlledValue = (value: number | null): any => {
  return value === null ? "" : value;
};

const getNumericValue = (value: string, options: any): number | null => {
  const { integer, minValue, maxValue } = options;
  const numericValue = integer ? parseInt(value, 10) : parseFloat(value);
  if (isNaN(numericValue)) {
    return null;
  }

  if (numericValue >= minValue && numericValue <= maxValue) {
    return numericValue;
  }

  return numericValue < minValue ? minValue : maxValue;
};

export const DATA_CY_DEFAULT = "input-number";
export const DATA_CY_SHORTCUT = "label";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [
  { name: "label", type: "string" },
  { name: "placeholder", type: "string" },
];

const InputNumber: FC<IInputNumber> = ({
  dataCy = DATA_CY_DEFAULT,
  disabled = false,
  integer = true,
  label,
  minValue = Number.MIN_SAFE_INTEGER,
  maxValue = Number.MAX_SAFE_INTEGER,
  onChange = undefined,
  placeholder = undefined,
  required = false,
  shrink = true,
  size = InputSize.default,
  value = null,
  variant = InputVariant.default,
}) => {
  const onChangeHandler = (event: any) => {
    const numericValue = getNumericValue(event.target.value, { integer, minValue, maxValue });
    if (onChange) {
      onChange(numericValue);
    }
  };

  return (
    <StyledMUITextField
      disabled={disabled}
      InputLabelProps={{
        shrink,
      }}
      inputProps={{
        "data-cy": dataCy,
        style: { textAlign: "right" },
      }}
      InputProps={{
        readOnly: disabled,
      }}
      label={label}
      margin="normal"
      onChange={onChangeHandler}
      placeholder={placeholder}
      required={required}
      size={size}
      type={InputDataType.number}
      variant={variant}
      value={getControlledValue(value)}
    />
  );
};

export const InputNumberWithProps = InputNumber;

export default localized(InputNumber, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});
