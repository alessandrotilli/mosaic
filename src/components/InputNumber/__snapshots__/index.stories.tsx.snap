// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs/InputNumber Adornment smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r1:"
         id=":r1:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r1:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              tabindex="-1"
              type="button"
              disabled
              data-cy="icon-button"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-e5efhf-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CloseIcon"
             data-cy="icon-button-icon"
        >
          <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber AdornmentClickable smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r2:"
         id=":r2:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r2:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
              data-cy="icon-button"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-e5efhf-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CloseIcon"
             data-cy="icon-button-icon"
        >
          <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
          </path>
        </svg>
        <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
        </span>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Decimal smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r3:"
         id=":r3:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r3:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value="5.275"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Disabled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r4:"
         id=":r4:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           disabled
           id=":r4:"
           readonly
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-readOnly MuiInputBase-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Localized smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r5:"
         id=":r5:-label"
  >
    Campo numerico
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r5:"
           placeholder="Inserire valore..."
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Campo numerico
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber MinMax smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r6:"
         id=":r6:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r6:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Nullable smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r7:"
         id=":r7:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r7:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Placeholder smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r8:"
         id=":r8:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r8:"
           placeholder="Enter value..."
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Primary smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r0:"
         id=":r0:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r0:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Required smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r9:"
         id=":r9:-label"
  >
    Label
    <span aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
    >
       *
    </span>
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":r9:"
           required
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label *
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber SizeSmall smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":ra:"
         id=":ra:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":ra:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Styled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":rb:"
         id=":rb:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           id=":rb:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value="5"
           style="text-align: center; width: 100%; color: red; font-weight: bold; font-size: large;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber VariantFilled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":rc:"
         id=":rc:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ff8729-MuiInputBase-root-MuiFilledInput-root">
    <input aria-invalid="false"
           id=":rc:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
  </div>
</div>
`;

exports[`Inputs/InputNumber VariantStandard smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":rd:"
         id=":rd:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root">
    <input aria-invalid="false"
           id=":rd:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
           value="5"
           style="text-align: right; width: 100%;"
    >
  </div>
</div>
`;
